backend/lib/spark-api-deal-submitter.js

```diff
@@ -0,0 +1,127 @@
+/** @import {PgPool, Queryable} from '@filecoin-station/deal-observer-db' */
+import Cursor from 'pg-cursor'
+
+/**
+ * Finds deals that haven't been submitted to the Spark API yet and submits them.
+ *
+ * @param {PgPool} pgPool
+ * @param {number} batchSize
+ * @param {(eligibleDeals: Array) => Promise<void>} submitDeals
+ */
+export const findAndSubmitUnsubmittedDeals = async (pgPool, batchSize, submitDeals) => {
+  console.debug(`Finding and submitting deals using batchSize: ${batchSize}`)
+  for await (const unsubmittedDeals of findUnsubmittedDeals(pgPool, batchSize)) {
+    console.debug(`Found ${unsubmittedDeals.length} unsubmitted deals`)
+    try {
+      const formattedDeals = formatUnsubmittedDealsForSparkApi(unsubmittedDeals)
+      await submitDeals(formattedDeals)
+      console.debug(`Successfully submitted ${formattedDeals.length} deals`)
+      await markDealsAsSubmitted(pgPool, unsubmittedDeals)
+    } catch (e) {
+      console.error('Failed to submit deals:', e)
```

Comment:
> In the case of an error during submission we won't retry to submit deals again. Rather submission loop will pick up unsubmitted deals next time it runs and will try to post them again.

If it's reasonably easy, then please consider adding a unit-test to verify this behaviour.

Reply:
resolving due to lack of time